### ISSUE TITLE
follow up 101 (task icons)

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-ui#109]
From bottom of description in #101 

(We should either replace this issue with multiple separate issues, or turn the following into a 
check-point list and address with multiple PRs)

TODO (future work)
- [x] Incorporate into views.
   - Work out how we are going to feed data into these components (e.g. props or whatever).
- [ ] Animate state changes (see #127)
   -  Pure CSS probably wont be enough.
   - [edit] Actually CSS is probably the best bet, we are now using CSS animations to set the clock-face position.
   - More below...
- [x] Add the more eccentric task state~~s~~ (expired, ~~queued, retrying~~) and the held badge.
   - Expired and queued are easy, the rest will require a dig into the Cylc Flow code to see what separation is possible with the current internals of Cylc.
- [x] (#134) Extract the <style> from the Job component to make it theme-able.

**Comment by kinow** [comment]
I think I can raise a PR for the first item about props.

**Comment by kinow** [comment]
>I think I can raise a PR for the first item about props.

Done: https://github.com/cylc/cylc-ui/pull/110

**Comment by oliver-sanders** [comment]
Support for held and retrying states is blocked by:

* https://github.com/cylc/cylc-flow/issues/3223

**Comment by hjoliver** [comment]
Suggested icon design for `expired` task state (carried forward from #201):
![image](https://user-images.githubusercontent.com/2362137/68527136-988b0e00-0348-11ea-9f51-1d25fb6ad2db.png)

**Comment by oliver-sanders** [comment]
The last thing on this issue is animation of state changes.

It might be useful to highlight state changes to make them more obvious to the user. IIn particular changes to the `failed` and `submit-failed` states.

Any animations would need to be minimal and subtle to avoid becoming more hindrance than help.

We are using CSS animations to set the clock-face position as the task is running. We have had to limit the number of frames for performance reasons so animations should be short lived and easy to compute too.

Suggest a quick, simple size or colour change.

**Comment by oliver-sanders** [comment]
The last point is covered by #127 so closing this umbrella issue.
